src/appsignal/cli/diagnose.py

```diff
@@ -0,0 +1,382 @@
+
+    def working_directory_user_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["uid"]["result"]
+        )
+
+    def working_directory_group_id(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["gid"]["result"]
+        )
+
+    def working_directory_permissions(self) -> bool:
+        return (
+            "working_directory_stat" in self.report
+            and self.report["working_directory_stat"]["mode"]["result"]
+        )
+
+    def lock_path(self) -> str:
+        if "lock_path" in self.report and self.report["lock_path"]["created"]["result"]:
+            return "writable"
+        return "not writable"
+
+
+class DiagnoseCommand(AppsignalCLICommand):
+    @staticmethod
+    def init_parser(parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "--send-report",
+            action="store_true",
+            help="Send the report to AppSignal",
+        )
+        parser.add_argument(
+            "--no-send-report",
+            action="store_true",
+            help="Do not send the report to AppSignal",
+        )
+
+    def run(self) -> int:
+        agent = Agent()
+        agent_json = json.loads(agent.diagnose())
+        self.config = Config()
+        self.send_report = self.args.send_report
+        self.no_send_report = self.args.no_send_report
+        self.agent_report = AgentReport(agent_json)
+
+        self.report = {
+            "agent": {
+                "agent": {
+                    "boot": agent_json["boot"],
+                    "config": {
+                        "valid": agent_json["config"]["valid"],
+                    },
+                    "host": agent_json["host"],
+                    "lock_path": agent_json.get("lock_path"),
+                    "logger": agent_json.get("logger"),
+                    "working_directory_stat": agent_json.get("working_directory_stat"),
+                }
+            },
+            "config": {
+                "options": self.config.options,
+                "sources": self.config.sources,
+            },
+            "host": {
+                "architecture": platform.machine(),
+                "heroku": os.environ.get("DYNO") is not None,
+                "language_version": platform.python_version(),
+                "os": platform.system().lower(),
+                "os_distribution": self._os_distribution(),
+                "root": os.getuid() == 0,
+                "running_in_container": self.config.option("running_in_container")
+                is not None,
+            },
+            "library": {
+                "language": "python",
+                "package_version": __version__,
+                "agent_version": "91f1a7c",
+            },
+            "paths": self._paths_data(),
+            "process": {
+                "pid": os.getpid(),
```

Comment:
```suggestion
                "uid": os.getuid(),
```
We want to track the user id here.